
<launch>

	<include file="$(find jackal_navigation)/launch/include/rbcar_planning_mapping.launch" />

	<!--  ************** Gazebo Simulator ***************  -->
	<include file="$(find jackal_gazebo)/launch/parking_world_rb_mapping.launch" />
	<node pkg="abspos_rbcar" type="abspos_rbcar.py" name="abs_pos" />
	<node pkg="teb_local_planner_tutorials" type="cmd_vel_to_ackermann_drive.py" name="cmd_vel_to_ackermann" />


	
	  <!-- ************** Navigation ***************   -->
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
  	  	<rosparam file="$(find teb_local_planner_tutorials)/cfg/carlike/costmap_common_params_dynamic.yaml" command="load" ns="global_costmap" />
  	 	<rosparam file="$(find teb_local_planner_tutorials)/cfg/carlike/costmap_common_params_dynamic.yaml" command="load" ns="local_costmap" />
  		<rosparam file="$(find teb_local_planner_tutorials)/cfg/carlike/local_costmap_params_dynamic.yaml" command="load" />
  		<rosparam file="$(find teb_local_planner_tutorials)/cfg/carlike/global_costmap_params_dynamic.yaml" command="load" />
  		<rosparam file="$(find teb_local_planner_tutorials)/cfg/carlike/teb_local_planner_params_dynamic.yaml" command="load" />
  	

  		<!-- This is the ROS-inbuilt global planner -->
		<!-- <param name="base_global_planner" value="global_planner/GlobalPlanner" /> -->

		<!-- This is the planner I am testing -->
		<param name="base_global_planner" value="lattice_planner/LatticePlannerROS" /> 


		<param name="planner_frequency" value="1.0" />
		<param name="planner_patience" value="5.0" />

		<!-- Teb local planner -->
		<param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />

		<!-- Local Planner from Jackal -->
		<!-- <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/>  -->

		<param name="controller_frequency" value="10.0" />
		<param name="controller_patience" value="15.0" />

                <param name="clearing_rotation_allowed" value="false" /> 
	</node>


	<!--  **************** Visualisation ****************  -->
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find teb_local_planner_tutorials)/cfg/rviz_navigation.rviz"/>

	<!-- <include file="$(find jackal_viz)/launch/view_robot.launch" /> -->

</launch>